EIPS/eip-2935.md

```diff
@@ -27,15 +27,19 @@ Additional secondary motivations include:
 
 | Parameter | Value |
 | - | - |
-| `FORK_BLKTIME` | TBD |
+| `FORK_TIMESTAMP` | TBD |
 | `HISTORY_STORAGE_ADDRESS` | `0xfffffffffffffffffffffffffffffffffffffffe`|
+| `HISTORY_SERVE_WINDOW` | `256` |
 
-At the start of processing any block where `block.timestamp > FORK_BLKTIME` (ie. before processing any transactions), run `sstore(HISTORY_STORAGE_ADDRESS, block.number - 1, block.prevhash)`.
-
-When `sload(HISTORY_STORAGE_ADDRESS, max(block.number - 256, 0)) != 0`, change the logic of the `BLOCKHASH` opcode as follows: return `sload(HISTORY_STORAGE_ADDRESS, arg)`.
-
-Note that this means that slot `max(block.number - 256, 0)` is accessed during block execution.
+At the start of processing any block where `block.timestamp > FORK_TIMESTAMP` (ie. before processing any transactions), 
+* run `sstore(HISTORY_STORAGE_ADDRESS, block.number - 1, block.prevhash)`. 
+* additionally if this is the first block of the fork, then store last `min(HISTORY_SERVE_WINDOW-1, block.number-1)` blocks' prevhash into their corresponding slots i.e persist the entire `HISTORY_SERVE_WINDOW` block hashes (up until the genesis) that can be accessed in the block.
```

Comment:
so actually it would need to be `min(HISTORY_SERVE_WINDOW, block.number)` since the genesis block would have to be inserted as well, and you need to insert all 256 previous blocks at fork block time.

Reply:
yes there are additional insert's of blocks' prevhash

Reply:
true, then maybe put "additionally" in bold so that people like me notice.